File: contains-duplicate/SamTheKorean.go
// TC : O(n) : it iterates for the length of nums
// SC : O(n) : hashmap is created with the size of nums

func containsDuplicate(nums []int) bool {
    hashmap := make(map[int]bool)

    for i:=0;i<len(nums);i++ {
        if hashmap[nums[i]] {
            return true
        }

        hashmap[nums[i]] = true
    }

    return false
}

File: contains-duplicate/SamTheKorean.py
# TC : O(n) n being a length of nums array and it iterates the nums array once
# SC : O(n) n being a size of nums array
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False
